lib/dust.js

```diff
@@ -861,6 +861,17 @@
     var body = bodies.block,
         skip = bodies['else'];
 
+    if (dust.isThenable(elem)) {
```

Comment:
Could you turn this into a little reusable func that we can use for exists and notexists both? Look above at `getWithResolvedData` as an example-- just a little helper func not attached to the Chunk prototype.